lemur/plugins/lemur_kubernetes/plugin.py

```diff
@@ -106,33 +125,38 @@ def __init__(self, *args, **kwargs):
 
     def upload(self, name, body, private_key, cert_chain, options, **kwargs):
 
-        k8_bearer = self.get_option('kubernetesAuthToken', options)
-        k8_cert = self.get_option('kubernetesServerCertificate', options)
-        k8_namespace = self.get_option('kubernetesNamespace', options)
-        k8_base_uri = self.get_option('kubernetesURL', options)
-
-        k8s_api = K8sSession(k8_bearer, k8_cert)
-
-        cert = Certificate(body=body)
-
-        # in the future once runtime properties can be passed-in - use passed-in secret name
-        secret_name = 'certs-' + urllib.quote_plus(cert.name)
-
-        err = ensure_resource(k8s_api, k8s_base_uri=k8_base_uri, namespace=k8_namespace, kind="secret", name=secret_name, data={
-            'apiVersion': 'v1',
-            'kind': 'Secret',
-            'metadata': {
-                'name': secret_name,
-            },
-            'data': {
-                'combined.pem': base64.b64encode(body + private_key),
-                'ca.crt': base64.b64encode(cert_chain),
-                'service.key': base64.b64encode(private_key),
-                'service.crt': base64.b64encode(body),
-            }
-        })
+        try:
+            k8_bearer = self.get_option('kubernetesAuthToken', options)
+            k8_cert = self.get_option('kubernetesServerCertificate', options)
+            k8_namespace = self.get_option('kubernetesNamespace', options)
+            k8_base_uri = self.get_option('kubernetesURL', options)
+
+            k8s_api = K8sSession(k8_bearer, k8_cert)
+
+            cn = common_name(parse_certificate(body))
+
+            # in the future once runtime properties can be passed-in - use passed-in secret name
+            secret_name = 'certs-' + cn
+
+            err = ensure_resource(k8s_api, k8s_base_uri=k8_base_uri, namespace=k8_namespace, kind="secret", name=secret_name, data={
+                'apiVersion': 'v1',
+                'kind': 'Secret',
+                'metadata': {
+                    'name': secret_name,
+                },
+                'data': {
+                    'combined.pem': base64encode('%s\n%s' % (body, private_key)),
+                    'ca.crt': base64encode(cert_chain),
+                    'service.key': base64encode(private_key),
+                    'service.crt': base64encode(body),
+                }
+            })
+        except Exception as e:
+            current_app.logger.exception("Exception in upload")
+            raise e
```

Comment:
You can just say `raise` instead of `raise e`

Reply:
Excelent